flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/ExecutionFailureHandler.java

```diff
@@ -141,6 +155,51 @@ private CompletableFuture<Map<String, String>> labelFailure(Throwable cause, boo
         return FailureEnricherUtils.labelFailure(cause, ctx, mainThreadExecutor, failureEnrichers);
     }
 
+    private FailureHandlingResult handleFailureAndReport(
+            @Nullable final Execution failedExecution,
+            final Throwable cause,
+            long timestamp,
+            final Set<ExecutionVertexID> verticesToRestart,
+            final boolean globalFailure) {
+
+        FailureHandlingResult failureHandlingResult =
+                handleFailure(failedExecution, cause, timestamp, verticesToRestart, globalFailure);
+
+        if (reportEventsAsSpans) {
+            // TODO: replace with reporting as event once events are supported.
+            // Add reporting as callback for when the failure labeling is completed.
+            failureHandlingResult
+                    .getFailureLabels()
+                    .thenAcceptAsync(
+                            labels -> reportFailureHandling(failureHandlingResult, labels),
+                            mainThreadExecutor);
+        }
+
+        return failureHandlingResult;
+    }
+
+    private void reportFailureHandling(
+            FailureHandlingResult failureHandlingResult, Map<String, String> failureLabels) {
+
+        // Add base attributes
+        SpanBuilder spanBuilder =
+                Span.builder(ExecutionFailureHandler.class, "JobFailure")
+                        .setStartTsMillis(failureHandlingResult.getTimestamp())
+                        .setEndTsMillis(failureHandlingResult.getTimestamp())
+                        .setAttribute(
+                                "canRestart", String.valueOf(failureHandlingResult.canRestart()))
+                        .setAttribute(
+                                "isGlobalFailure",
+                                String.valueOf(failureHandlingResult.isGlobalFailure()));
```

Comment:
Can you add this span to the documentation of reported spans?

Reply:
That's what I had before, but since it's not enabled by default right now I'd suggest we document this once it is reported as event by default.